marketplacetests/marketplace/pages/base.py

```diff
@@ -0,0 +1,179 @@
+        from marketplacetests.marketplace.pages.search_results import SearchResults
+        return SearchResults(self.marionette)
+
+    def set_region(self, region):
+        self._perform_search(':debug')
+        from marketplacetests.marketplace.pages.debug import Debug
+        debug_screen = Debug(self.marionette)
+        debug_screen.select_region(region)
+        # wait for notification of the change
+        self.wait_for_region_updated_notification(region)
+
+        debug_screen.tap_back()
+        self.wait_for_page_loaded()
+
+    def navigate_to_app(self, app_name):
+        search_results = self.search(app_name).search_results
+        for result in search_results:
+            if result.name == app_name:
+                return result.tap_app()
+
+        # app not found
+        raise Exception('The app: %s was not found.' % app_name)
+
+    def tap_back(self):
+        back_button = self.marionette.find_element(*self._back_button_locator)
+        # This workaround is required for gaia v2.0, but can be removed in later versions
+        # as the bug has been fixed
+        # Bug 937053 - tap() method should calculate elementInView from the coordinates of the tap
+        self.marionette.execute_script('arguments[0].scrollIntoView(false);', [back_button])
+        back_button.tap()
+
+    def tap_home(self):
+        self.marionette.find_element(*self._home_button_locator).tap()
+
+    @property
+    def install_notification_message(self):
+        self.marionette.switch_to_frame()
+        self.wait_for_element_displayed(*self._install_notification_locator)
+        return self.marionette.find_element(*self._install_notification_locator).text
+
+    @property
+    def nav_menu(self):
+        return NavMenu(self.marionette)
+
+
+class NavMenu(Marketplace):
+
+    _nav_menu_toggle_locator = (By.CSS_SELECTOR, 'mkt-nav-toggle button')
+    _categories_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[title="Categories"]')
+    _feedback_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[href*="feedback"]')
+    _home_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[title="Home"]')
+    _my_apps_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[href*="purchases"]')
+    _new_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[href*="new"]')
+    _popular_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[href*="popular"]')
+    _settings_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link[href*="settings"]')
+    _sign_in_menu_item_locator = (By.CSS_SELECTOR, '.mkt-nav--link.persona:not(.register)')
+
+    def open(self):
+        body = self.marionette.find_element(By.TAG_NAME, 'body')
+        if not 'mkt-nav--visible' in body.get_attribute('class'):
+            self.marionette.find_element(*self._nav_menu_toggle_locator).tap()
+            Wait(self.marionette).until(
+                lambda m: 'mkt-nav--visible' in body.get_attribute('class'))
+
+    def tap_settings(self):
+        self.open()
```

Comment:
With the above suggested change, we could remove this line and expect the test to explicitly open the menu first.